### PR TITLE
Move codeReviewEnabled out of sectionData redux

### PR DESCRIPTION
**Pull request by maureensturgeon**
This is part of a larger effort to consolidate `teacherSectionsRedux` and `sectionDataRedux`. In this PR I've moved a couple of things out of `sectionDataRedux`
1. `codeReviewEnabled` which is used on a level to determine if the user should see the code review experience, this has been moved to instructions redux and renamed to `codeReviewEnabledForLevel` since at this point the data represents more than just the section level setting (whether the dcdo flag is on, the type of level, whether the user is in code review groups etc)
2. I've moved `codeReviewExpiresAt` to `teacherSectionsRedux`, this data _does_ represent a section-level setting and `teacherSectionsRedux` is populated for the page where this field is used, so it will live there with the rest of the section-related data

## Links
- jira ticket: [LP-2179](https://codedotorg.atlassian.net/browse/LP-2179)

## Testing story
Tested locally that manage students enable and disable was working as expected

## Follow-up work
Remove `sectionData` redux and replace with `teacherSectionsRedux`

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/sites/studio/pages/levels/show.js [hunk 1]
```diff
@@ -18,13 +16,13 @@ function initPage() {
   const script = document.querySelector('script[data-level]');
   const config = JSON.parse(script.dataset.level);
 
-  registerReducers({sectionData, instructions});
+  registerReducers({instructions});
```
Comment (jamescodeorg) [inline apps/src/sites/studio/pages/levels/show.js:19]: Nice!

**Review by jamescodeorg — APPROVED** [review]
Curious to know what @bencodeorg thinks, but these look like great changes to me!

File: dashboard/app/controllers/script_levels_controller.rb [hunk 2]
```diff
@@ -510,18 +510,15 @@ def present_level
       )
     end
 
-    # To do: rename this variable, as it gets passed into redux as sectionData.section.codeReviewEnabled,
```
Comment (bencodeorg) [inline dashboard/app/controllers/script_levels_controller.rb:513]: Thank you so much for doing this!

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsStatusToggle.jsx [hunk 3]
```diff
@@ -80,12 +83,12 @@ export const UnconnectedCodeReviewGroupsStatusToggle = CodeReviewGroupsStatusTog
 
 export default connect(
   state => ({
-    codeReviewExpiresAt: state.sectionData.section.codeReviewExpiresAt,
-    sectionId: state.sectionData.section.id
+    codeReviewExpiresAt: selectedSection(state).codeReviewExpiresAt,
```
Comment (bencodeorg) [inline apps/src/templates/codeReviewGroups/CodeReviewGroupsStatusToggle.jsx:86]: Nit: seems unlikely you'd hit this, but could you end up trying to read properties of null here?

**Review by bencodeorg — APPROVED** [review]
Looks great!

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsStatusToggle.jsx (hunk 3 above)
Comment (maureensturgeon) [reply apps/src/templates/codeReviewGroups/CodeReviewGroupsStatusToggle.jsx:86]: This component is on the manage students page, a section needs to be selected to end up on this page, so no, I don't believe you'd be able to end up with a null selected section

**Pull request merged**